From 62df32d604fb95243b0eaf1db8a8ef00d060ea1e Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Wed, 7 Nov 2018 11:09:02 -0200
Subject: [PATCH 162/174] support: Fix printf format for TEST_COMPARE_STRING

Fix the following on 32 bits targets:

support_test_compare_string.c: In function 'support_test_compare_string':
support_test_compare_string.c:80:37: error: format '%lu' expects argument of
type 'long unsigned int', but argument 2 has type 'size_t' {aka 'unsigned int'}
[-Werror=format=]
         printf ("  string length: %lu bytes\n", left_length);
                                   ~~^           ~~~~~~~~~~~
                                   %u
Checked on arm-linux-gnueabihf.

	* support/support_test_compare_string.c
	(support_test_compare_string): Fix printf format.

(cherry picked from commit 00c86a37d1b63044e3169d1f2ebec23447c73f79)
---
 ChangeLog                             | 5 +++++
 support/support_test_compare_string.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/ChangeLog b/ChangeLog
index ba53136bec..190b9ea0d0 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2018-11-07  Adhemerval Zanella  <adhemerval.zanella@linaro.org>
+
+	* support/support_test_compare_string.c
+	(support_test_compare_string): Fix printf format.
+
 2018-11-07  Florian Weimer  <fweimer@redhat.com>
 
 	Implement TEST_COMPARE_STRING.
diff --git a/support/support_test_compare_string.c b/support/support_test_compare_string.c
index 9958aaeec1..a76ba8eda7 100644
--- a/support/support_test_compare_string.c
+++ b/support/support_test_compare_string.c
@@ -77,7 +77,7 @@ support_test_compare_string (const char *left, const char *right,
       support_record_failure ();
       printf ("%s:%d: error: blob comparison failed\n", file, line);
       if (left_length == right_length && right != NULL && left != NULL)
-        printf ("  string length: %lu bytes\n", left_length);
+        printf ("  string length: %zu bytes\n", left_length);
       else
         {
           report_length ("left", left, left_length);
-- 
2.30.2

